puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -203,6 +214,18 @@ def resource_status_to_hash(resource_status)
     }
   end
 
+  # Get the size of the columns that will populate the resource_events_resource_timestamp_xxxxxz index.
+  # resource_events_resource_timestamp_xxxxxz index is comprised of 3 columns (resource timestamp, title and type)
+  # and it's maximum size is 2704 bytes. So the size of the 3 resource attributes combined, must not exceed 2704 bytes.
+  #
+  # @return Boolean
+  # @api private
+  def resource_event_index_size(resource_status)
+    total_size = Puppet::Util::Puppetdb.to_wire_time(resource_status.time).size +
```

Comment:
See the [calculation comment](https://github.com/puppetlabs/puppetdb/pull/3541#discussion_r682983482) on the Clojure PR for more info, but I think we need a similar workflow here like
```
total_size = 8 + resource_status.resource_type.bytesize + resource_status.title.to_s.bytesize

Reply:
Updated. Thanks!